### PR TITLE
Feature/shorten url

### PR DESCRIPTION
**Pull request by wykhuh**
@mikehmorrissey @pigstake @nrswolf Use slugs instead of random id for board url. Click button to copy board to clipboard.

Since we are using slugs for the board id, the board name is required.  How do I make a field required in Meteor besides setting the input to required?

The copy to clipbboard feature requires flash. Can someone with a mobile device check if that feature works if a device does not allow flash?

### CODE REVIEW

**Comment by mikehmorrissey** [comment]
@wykhuh the best way to ensure uniqueness is to create a unique index at the db level. Meteor doesn't have great documentation around this but the current way to achieve this is using ```._ensureIndex```. in this case it would look like: 
```js
Boards._ensureIndex({ slug: '1' }, { unique: true })
```

I feel like the friendly-slugs package might already take of this though.

**Comment by mikehmorrissey** [comment]
@wykhuh looks like copy to clipboard won't work w/ devices that don't support flash. I think that's fine considering the use case. we should just hide the button if flash isn't supported. what do you think @pigstake?

**Comment by mikehmorrissey** [comment]
@wykhuh looks great! a couple comments/suggestions but beyond that it looks ready to :package:

**Comment by pigstake** [comment]
@wykhuh @mikehmorrissey Do we have Modernizr or some feature detection method for targeting when flash isn't supported?

**Comment by wykhuh** [comment]
@mikehmorrissey  I was asking about making a form field required; not uniqueness. I want  board title to be required when users fill out the form. I set the input to required, but that doesn't work on all browsers. 

  FYI, the slug package does check for uniqueness.

**Comment by mikehmorrissey** [comment]
@wykhuh ah okay. the other way would be to check the length of the value in that input before submitting. i'm not aware of a blaze way to do it but i haven't looked too deeply

**Comment by wykhuh** [comment]
@mikehmorrissey  added collection2 package to specify field schem, and do form validation.

File: client/modules/new-board/new-board.html [hunk 1]
```diff
@@ -4,8 +4,11 @@
       <h1 class="pageHeading">Time to vote?</h1>
       <h2 class="pageSubHeading">Create a board. <i>Go on.</i> You know you want to.</h2>
     </header>
+    {{ #if errors }}
```
Comment (wykhuh) [inline client/modules/new-board/new-board.html]: @pigstake Added an error messages if the form is invalid. Need some css help.

**Comment by pigstake** [comment]
@wykhuh Can you rebase with master if that's the conflict and merge this?

**Pull request merged**